src/structures/PermissionOverwrites.js

```diff
@@ -48,33 +51,20 @@ class PermissionOverwrites {
   }
 
   /**
-   * Updates this permissionOverwrites.
+   * Edits this Permission Overwrite.
    * @param {PermissionOverwriteOptions} options The options for the update
    * @param {string} [reason] Reason for creating/editing this overwrite
    * @returns {Promise<PermissionOverwrites>}
    * @example
    * // Update permission overwrites
-   * permissionOverwrites.update({
+   * permissionOverwrites.edit({
    *   SEND_MESSAGES: false
    * })
    *   .then(channel => console.log(channel.permissionOverwrites.get(message.author.id)))
    *   .catch(console.error);
    */
-  async update(options, reason) {
-    const { allow, deny } = this.constructor.resolveOverwriteOptions(options, this);
-
-    await this.channel.client.api
-      .channels(this.channel.id)
-      .permissions(this.id)
-      .put({
-        data: {
-          id: this.id,
-          type: OverwriteTypes[this.type],
-          allow,
-          deny,
-        },
-        reason,
-      });
+  async edit(options, reason) {
+    await this.channel.permissionOverwrites.edit(this.id, options, reason);
```

Comment:
Time to skip checks! Call permissionOverwrites.upsert with the `id, options, { type: this.type (but number), reason }, this`